src/fdb5/api/FDB.h

```diff
@@ -72,73 +76,200 @@ class FDB {
     void archive(const metkit::mars::MarsRequest& request, eckit::DataHandle& handle);
 
-    // disclaimer: this is a low-level API. The provided key and the corresponding data are not checked for consistency
-    // Optional callback function is called upon receiving field location from the store.
+    /// Archive binary data to a FDB.
+    /// Any callback set with registerArchiveCallback will be invoked.
+    /// @note No constistency checks are applied. The caller needs to ensure the provided key matches metadata present
+    /// in data.
+    /// @param key Key used for indexing and archiving the data
+    /// @param data Pointer to the binary data to archive
+    /// @param length Size of the data to archive with the given @p key
     void archive(const Key& key, const void* data, size_t length);
 
+    /// Generate an new index entry for an existing field location.
+    /// Can be used to reindex existing data into a new catalogue (see fdb-reindex tool).
+    /// @param key Key used to index the data.
+    /// @param location Location of existing data in an FDB store.
     void reindex(const Key& key, const FieldLocation& location);
 
-    /// Flushes all buffers and closes all data handles into a consistent DB state
+    /// Flush all buffers and closes all data handles into a consistent DB state
     /// @note always safe to call
     void flush();
 
+    /// Read binary data from an URI.
+    /// @param uri eckit uri to the data source
+    /// @return DataHandle for reading the requested data from
     eckit::DataHandle* read(const eckit::URI& uri);
 
-    eckit::DataHandle* read(const std::vector<eckit::URI>& uris, bool sorted = false);
-
-    eckit::DataHandle* read(ListIterator& it, bool sorted = false);
-
+    /// Read binary data from an list of URI.
+    /// @param vector of uris eckit uris to the data source
+    /// @param inStorageOrder if set data will be returned in the order it is stored. If unset data will be returned in
+    /// the order it was requested.
+    /// @return DataHandle for reading the requested data
+    eckit::DataHandle* read(const std::vector<eckit::URI>& uris, bool inStorageOrder = false);
+
+    /// Read binary from a ListIterator.
+    /// @param uris a list iterator which resembles a set of fields which should be read
+    /// @param inStorageOrder if set data will be returned in the order it is stored. If unset data will be returned in
+    /// the order it was requested.
+    /// @return DataHandle for reading the requested data from
+    eckit::DataHandle* read(ListIterator& it, bool inStorageOrder = false);
+
+    /// Retrieve data which is specified by a MARS request.
+    /// @param request MarsRequest which describes the data which should be retrieved
+    /// @return DataHandle for reading the requested data from
     eckit::DataHandle* retrieve(const metkit::mars::MarsRequest& request);
 
+    // TODO(kkratz): Provide doc!
```

Comment:
Similar to list, but walks the schema rather than walking the contents.

Optimised for small retrieve against a large dataset. Used as part of the internal of the retrieve() call.